### DISCUSSION TITLE
How to correctly instrument library distributed as NuGet package?

### DISCUSSION THREAD

**Discussion by valxv** [open-telemetry/opentelemetry-dotnet#3919] [Q&A] [answered]
If `ActivitySource` created in my library has a different name than in the calling application, it is ignored and does not provide any output.
In the Microservice example, you use Propagators API to propagate calling activity context from Producer to Consumer. But for some reason, it does not work in my case. For example:
```csharp
var serviceName = "App";
var serviceVersion = "0.0.1";

using var tracerProvider = Sdk.CreateTracerProviderBuilder()
       .AddSource(serviceName)
       .SetResourceBuilder(
           ResourceBuilder.CreateDefault()
               .AddService(serviceName: serviceName, serviceVersion: serviceVersion))
       .AddConsoleExporter()
       .Build();

using var appSource = new ActivitySource(serviceName, serviceVersion);
using var appActivity = appSource.StartActivity("App activity", ActivityKind.Producer);
if (appActivity?.IsAllDataRequested == true)
{
       using var librarySource = new ActivitySource("Some Library name", serviceVersion);
       using var childActivity = librarySource.StartActivity("Library activity", ActivityKind.Consumer, appActivity.Context);
       childActivity?.AddEvent(new ActivityEvent("Test event"));
}
```
The `childActivity` becomes logged only if I initialize the `librarySource` with the same `serviceName`. If they have the same name I don't even need to pass parent activity context to the `librarySource.StartActivity` method.

Should I somehow pass to the library the calling application `ActivitySource` (`ActivitySource` name)? 
Is there a better way?

**Comment by cijothomas** [comment, marked as answer]
The app owner has to do AddSource("Some Library name"). 
AddSource() supports wild card, so a "*" will listen to *all* sources in the process.

**Reply by valxv** [reply]
Thanks!
